src/tags/include_tag.rs

```diff
@@ -0,0 +1,66 @@
+
+use Renderable;
+use context::Context;
+use token::Token;
+use LiquidOptions;
+use error::{Result, Error};
+
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+
+struct Include {
+    partial_path: String,
+}
+
+impl Renderable for Include {
+    fn render(&self, _context: &mut Context) -> Result<Option<String>> {
+        let path_str = self.partial_path.clone();
+        let path = Path::new(&path_str);
+
+        // check if file exists
+        if !path.exists() {
+            return Error::renderer(&format!("{:?} does not exist", path));
+        }
+
+        let mut file = match File::open(path) {
+            Ok(file) => file,
+            Err(e) => return Error::renderer(&format!("[std::io::Error] {}", e)),
+        };
+
+        let mut content = String::new();
+        match file.read_to_string(&mut content) {
+            Ok(_) => Ok(Some(content)),
+            Err(e) => Error::renderer(&format!("[std::io::Error] {}", e)),
+        }
+    }
+}
+
+pub fn include_tag(_tag_name: &str,
+                   arguments: &[Token],
+                   _options: &LiquidOptions)
+                   -> Result<Box<Renderable>> {
+    let mut args = arguments.iter();
+
+    match args.next() {
+        Some(&Token::Identifier(ref path)) => Ok(Box::new(Include { partial_path: path.clone() })),
```

Comment:
You need include the file, lex and parse it at compile time, since we don't want to do the lexing and parsing at render time. While this might not make a big difference to Cobalt, I've heard of people who want to use liquid on web servers, where render is called tons of times per minute.

"File not found" is a parser error, as strange as it sounds. Rust's libsyntax also does not successfully parse a file with incorrect `mod` statements. It makes sense if you consider that the `include`d file might also throw parser errors that we want to catch in the parsing step.

You don't have to use `Error::parse(expected, actual)`, you can make your own `Error::Parser` with a custom message.

You should basically copy what [liquid::parse](https://github.com/cobalt-org/liquid-rust/blob/master/src/lib.rs#L149) already does, without registering the tags and blocks and just passing the options through to the parser.

Then you can save the resulting `Template` in your struct and call `template.render(context)` at render time. :)